storage/innobase/dict/dict0stats.cc

```diff
@@ -2198,7 +2198,13 @@ dict_stats_update_persistent(
 
 	DEBUG_PRINTF("%s(table=%s)\n", __func__, table->name);
 
-	dict_table_stats_lock(table, RW_X_LATCH);
+	dict_table_analyze_index_lock(table);
```

Comment:
We cannot move it into dict_stats_analyze_index() because dict_stats_analyze_index() is called multiple times in dict_stats_update_persistent() and stat_sum_of_other_index_sizes_tmp is calculated. So we need to lock the whole process of analyzing of all indexes to get proper values.

Reply:
ok fair enough

Reply:
ok

Reply:
Can you also check the path where transient stats are calculated ? the split mutex should be used there as well

Reply:
For the given table we go even transient or persistent path (input parameter for dict_stats_update() set according to dict_stats_is_persistent_enabled(table) result). Transient path is fast, so it will not be a problem to lock statistics for the time of calculation.
It is not possible to go both paths in parallel for the same table.

Reply:
transient stats can use variable  innodb_stats_transient_sample_pages. If set to higher, I think we can see same issue. If you agree there is issue and can be fixed, it can be addressed as separate PR(please create JIRA ticket)

Reply:
Agree. Let's cover this scenario in separate PR related to PS-6148.

Reply:
thanks